### PR TITLE
Fix fetching from git

### PR DESCRIPTION
**Pull request by dcl10**
# Description
Previously, WfExS could only pull from specific remote sources; namely github, bitbucket and gitlab. However, self-hosted git repos may required in some instances. This PR removes the specificity and treats git as git.

This PR removes the capability to use `raw.githubusercontent.com` URLs to guess the repo. This is github specific and inherently fragile, because branches with slashes (`/`) in their names break this functionality. Use workflows as described by [PyPI](https://pip.pypa.io/en/stable/topics/vcs-support/#git) in `workflow_id`.

It is still possible to pull remote files individually using these urls, though.

### CODE REVIEW

**Comment by jmfernandez** [comment]
The only URLs I'm still allowing as repository links are https ones from repos, due historical reasons: bio.tools and WorkflowHub workflow contributors are used to tell their workflows through web URLs instead of more reliable permanent identifiers, as they are not used to them (and they are not accepted by command line tools like git)

**Comment by beforan** [comment]
@jmfernandez I take your point about some of the  ambiguous URLs, such as `ssh` URLs with a colon where that should be used to denote port, or pointing at the `.git` subdirectory.

Surely wfexs should support anything that the git client would, with something to denote that the intent is the URL points at a git repo. Then the pip convention of `git+<URL>` should suffice, and complying with the same behaviours as pip is desirable.

From our perspective, we just need a way to run a workflow from an airgapped environment. We do not have the luxury of pointing at `github.com` which is obviously by far the most common usage, because we cannot access the public internet.

We had hoped that `git+` URLs would allow us to checkout any git repository, whether locally on disk, or in a locally run service, e.g. GitLab running at `http://localhost:8080`.

However we found that currently wfexs reports an error when attempting to checkout from those URLs, as an "unsupported http(s) repository source" or similar.

The *intent* (though perhaps not the result) of this PR is to resolve that issue, allowing working support for `git+file://`, `git+http://`, `git+https://` and `git+ssh://` URLs regardless of hostname (which currently appears to only allow `github.com` and `gitlab.com`, with comments suggesting the addition of `bitbucket.org` in future...)

Can we work together to make that functionality work? Or otherwise make execution of a workflow from a local path e.g. `file:///absolute/path/to/my/workflow.cwl` work?

We are blocked until we have working workflow execution from a workflow source that is not a TRS endpoint, or a supported public git service (github, gitlab...), but the changes included in this PR unblocks us.

Obviously appreciate your input as you know the codebase much better than us, and the ramifications of any changes beyond what we see as fixes!

**Comment by beforan** [comment]
example workflow URLs we tried which failed with errors prior to the changes in this branch looked as follows:

- `git+file:///absolute/path/to/repo#subdirectory=relative/path/to/workflow.cwl`
- `git+http://localhost:8080/user/repo#subdirectory=relative/path/to/workflow.cwl`
    - where `localhost:8080` was running an instance of GitLab.

File: wfexs_backend/fetchers/git.py [hunk 1]
```diff
@@ -388,17 +389,29 @@ def fetch(
 HEAD_LABEL = b"HEAD"
 REFS_HEADS_PREFIX = b"refs/heads/"
 REFS_TAGS_PREFIX = b"refs/tags/"
+GIT_SCHEMES = ["https", "git+https", "ssh", "git+ssh", "file", "git+file"]
```
Comment (beforan) [inline wfexs_backend/fetchers/git.py:392]: I believe these should be limited to the `git+` forms only? As pure URLs of the schemes should not be handled by the git fetcher

**Comment by jmfernandez** [comment]
Hi, I'm going to merge your contributions in a new branch, so I can remediate some possible internal side effects

File: wfexs_backend/cwl_engine.py
Comment (jmfernandez) [inline wfexs_backend/cwl_engine.py:1]: This change is obsolete with the latest fixes where WfExS is going to use a fork of cwltool where I fixed the issue.

File: wfexs_backend/fetchers/git.py (hunk 1 above)
Comment (jmfernandez) [reply wfexs_backend/fetchers/git.py:392]: I agree

**Review by jmfernandez — APPROVED** [review]

**Pull request merged**